### DISCUSSION TITLE
Extending Snipe-IT pages with JavaScript

### DISCUSSION THREAD

**Discussion by SpeedyCraftah** [snipe/snipe-it#14619] [Q&A]
Hello,

What would be the best way to insert a custom JavaScript script into a specific Snipe-IT page (for instance, /hardware)?

For my use case I would like to modify the default filters for the column views, but want to avoid modifying the actual PHP code which would likely sabotage any update attempts, and since the filter preferences are stored in localstorage, an inserted JavaScript file sounds like a reasonable solution for the problem.

Thanks.

**Comment by snipe** [comment]
We don't allow custom javascript in Snipe-IT, as it would introduce security concerns. You would have to modify the JS or the PHP itself and manage updates.
